torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp

```diff
@@ -391,7 +391,7 @@ void InferShapeTypeForUninitializedOutput(
       const_node->t_(attr::value, at::zeros(size, elem_type));
       const_node->output()->setType(other_output->type());
     } else {
-      const_node->t_(attr::value, at::zeros({}, elem_type));
+      const_node->t_(attr::value, at::zeros(c10::SymIntArrayRef{}, elem_type));
```

Comment:
@ezyang @albanD 

I think the other thing we could do here is to add `at::zeros_autograd` that only takes `SymIntArrayRef` and keep `zeros` alone for now.

Reply:
which call sites would use this

Reply:
If there isn't a way to get this overload to work, I agree that it would be reasonable to move the overload to another C++ name to solve the problem. There is precedent for doing this with out overloads which have a similar problem. I'll construct a patch.

Reply:
try this

```
diff --git a/torchgen/api/cpp.py b/torchgen/api/cpp.py
index 39c3c8684fa..3532081b88f 100644
--- a/torchgen/api/cpp.py
+++ b/torchgen/api/cpp.py
@@ -66,6 +66,8 @@ def name(func: FunctionSchema, *, faithful_name_for_out_overloads: bool = False)
     name = str(func.name.name)
     if func.is_functional_fn():
         name += "_functional"
+    elif func.is_symint_fn():
+        name += "_symint"
     elif func.is_out_fn():
         if faithful_name_for_out_overloads:
             name += "_outf"
diff --git a/torchgen/model.py b/torchgen/model.py
index 052fc24bfd7..68d7f991166 100644
--- a/torchgen/model.py
+++ b/torchgen/model.py
@@ -1188,6 +1188,10 @@ class FunctionSchema:
     def is_functional_fn(self) -> bool:
         return "functional" in self.name.overload_name
 
+    def is_symint_fn(self) -> bool:
+        # TODO: make this more robust
+        return "SymInt" in self.name.overload_name
+
     def is_out_fn(self) -> bool:
         # Note [is_out_fn]
         #
```

and then fix any calls to expand that take symint to expand_symint

Reply:
needs some more codegen updates

Reply:
```
diff --git a/aten/src/ATen/native/TensorShape.cpp b/aten/src/ATen/native/TensorShape.cpp
index c1fc678921e..ff05b5f91fb 100644
--- a/aten/src/ATen/native/TensorShape.cpp
+++ b/aten/src/ATen/native/TensorShape.cpp
@@ -3394,7 +3394,7 @@ at::Tensor& diagonal_copy_out(const at::Tensor & self, int64_t offset, int64_t d
 
 
 at::Tensor& expand_copy_SymInt_out(const at::Tensor & self, c10::SymIntArrayRef size, bool implicit, at::Tensor & out) {
-  auto tmp = self.expand(size, implicit);
+  auto tmp = self.expand_symint(size, implicit);
   out.copy_(tmp);
   return out;
 }
diff --git a/tools/autograd/load_derivatives.py b/tools/autograd/load_derivatives.py
index feeecefc11a..0ff66d1ec7d 100644
--- a/tools/autograd/load_derivatives.py
+++ b/tools/autograd/load_derivatives.py
@@ -339,6 +339,10 @@ def postprocess_forward_derivatives(
                     arg_name = arg_name + "_t"
                 new_args.append(arg_name)
 
+            # TODO we are trolling
+            if f.func.is_symint_fn():
+                defn_name += "_symint"
+
             # Call into the forward again. We need two cases here to handle both Tensor methods and at:: functions.
             if Variant.function in f.variants:
                 fw_formula = "at::{}({})".format(defn_name, ", ".join(new_args))
@@ -396,7 +400,7 @@ def create_differentiability_info(
         functions: Sequence[NativeFunction], name: str
     ) -> NativeFunction:
         for f in functions:
-            if cpp.name(f.func) == name:
+            if not f.func.is_functional_fn() and not f.func.is_out_fn() and name == str(f.func.name.name):
                 return f
         # some functions only have in-place variants
         assert name + "_" == cpp.name(functions[0].func)
diff --git a/torch/csrc/lazy/core/shape_inference.cpp b/torch/csrc/lazy/core/shape_inference.cpp
index 4cc188e08f0..cbee60110f1 100644
--- a/torch/csrc/lazy/core/shape_inference.cpp
+++ b/torch/csrc/lazy/core/shape_inference.cpp
@@ -649,7 +649,7 @@ std::vector<Shape> compute_shape_repeat(const at::Tensor & self, at::IntArrayRef
   return {Shape(self.scalar_type(), target_size)};
 }
 
-std::vector<Shape> compute_shape_narrow_copy(const at::Tensor & self, int64_t dim, int64_t start, c10::SymInt length) {
+std::vector<Shape> compute_shape_narrow_copy_symint(const at::Tensor & self, int64_t dim, int64_t start, c10::SymInt length) {
   return {Shape(self.scalar_type(), self.sizes().vec())};
 }
 
diff --git a/torch/csrc/lazy/core/shape_inference.h b/torch/csrc/lazy/core/shape_inference.h
index 41477b7be60..c94b616f49f 100644
--- a/torch/csrc/lazy/core/shape_inference.h
+++ b/torch/csrc/lazy/core/shape_inference.h
@@ -74,7 +74,7 @@ TORCH_API std::vector<torch::lazy::Shape> compute_shape_sum(const at::Tensor & s
 TORCH_API std::vector<torch::lazy::Shape> compute_shape__to_copy(const at::Tensor & self, c10::optional<at::ScalarType> dtype, c10::optional<at::Layout> layout, c10::optional<at::Device> device, c10::optional<bool> pin_memory, bool non_blocking, c10::optional<at::MemoryFormat> memory_format);
 TORCH_API std::vector<torch::lazy::Shape> compute_shape_trace(const at::Tensor & self);
 TORCH_API std::vector<torch::lazy::Shape> compute_shape_zero_functional(const at::Tensor & self);
-TORCH_API std::vector<torch::lazy::Shape> compute_shape_narrow_copy(const at::Tensor & self, int64_t dim, int64_t start, c10::SymInt length);
+TORCH_API std::vector<torch::lazy::Shape> compute_shape_narrow_copy_symint(const at::Tensor & self, int64_t dim, int64_t start, c10::SymInt length);
 
 // Non-Native ops
 TORCH_API std::vector<Shape> compute_shape_scalar(const at::Scalar& value, const at::ScalarType& type);
diff --git a/torchgen/api/cpp.py b/torchgen/api/cpp.py
index 39c3c8684fa..44890c88277 100644
--- a/torchgen/api/cpp.py
+++ b/torchgen/api/cpp.py
@@ -66,7 +66,9 @@ def name(func: FunctionSchema, *, faithful_name_for_out_overloads: bool = False)
     name = str(func.name.name)
     if func.is_functional_fn():
         name += "_functional"
-    elif func.is_out_fn():
+    if func.is_symint_fn():
+        name += "_symint"
+    if func.is_out_fn():
         if faithful_name_for_out_overloads:
             name += "_outf"
         else:
diff --git a/torchgen/model.py b/torchgen/model.py
index 052fc24bfd7..68d7f991166 100644
--- a/torchgen/model.py
+++ b/torchgen/model.py
@@ -1188,6 +1188,10 @@ class FunctionSchema:
     def is_functional_fn(self) -> bool:
         return "functional" in self.name.overload_name
 
+    def is_symint_fn(self) -> bool:
+        # TODO: make this more robust
+        return "SymInt" in self.name.overload_name
+
     def is_out_fn(self) -> bool:
         # Note [is_out_fn]
         #